engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterFactory.java

```diff
@@ -48,93 +49,28 @@ public class WhereFilterFactory {
-        // <ColumnName>=<Number|Boolean|"String">
-        // <ColumnName>!=<Number|Boolean|"String">
+        // Each side may fit: (<ColumnName>|<QueryScopeParam>|<Number|Boolean|"String"|Literal>)
+        // Supported ops: ==, =, !=, <, <=, >, >=
         parser.registerFactory(new AbstractExpressionFactory<>(
-                START_PTRN + "(" + ID_PTRN + ")\\s*(?:(?:={1,2})|(!=))\\s*(" + LITERAL_PTRN + ")" + END_PTRN) {
+                START_PTRN + "(?:(" + ID_PTRN + ")|(" + LITERAL_PTRN + "))\\s*((?:=|!|<|>)=?)\\s*(?:(" + ID_PTRN + ")|("
+                        + LITERAL_PTRN + "))" + END_PTRN) {
             @Override
             public WhereFilter getExpression(String expression, Matcher matcher, Object... args) {
-                final String columnName = matcher.group(1);
-                final boolean inverted = matcher.group(2) != null;
-                final String value = matcher.group(3);
+                // LITERAL_PTRN has 5 groups; mostly non-capturing
+                final boolean leftIsId = matcher.group(1) != null;
+                final boolean rightIsId = matcher.group(8) != null;
 
-                final FormulaParserConfiguration parserConfiguration = (FormulaParserConfiguration) args[0];
-                if (isRowVariable(columnName)) {
-                    log.debug().append("WhereFilterFactory creating ConditionFilter for expression: ")
-                            .append(expression).endl();
-                    return ConditionFilter.createConditionFilter(expression, parserConfiguration);
+                if (!leftIsId && !rightIsId) {
+                    return ConditionFilter.createConditionFilter(expression, (FormulaParserConfiguration) args[0]);
                 }
-                log.debug().append("WhereFilterFactory creating MatchFilter for expression: ").append(expression)
-                        .endl();
-                return new MatchFilter(
-                        MatchFilter.CaseSensitivity.MatchCase,
-                        inverted ? MatchFilter.MatchType.Inverted : MatchFilter.MatchType.Regular,
-                        columnName,
-                        value);
-            }
-        });
-        // <ColumnName>==<User QueryScopeParam>
-        // <ColumnName>=<User QueryScopeParam>
-        // <ColumnName>!=<User QueryScopeParam>
-        parser.registerFactory(new AbstractExpressionFactory<>(
-                START_PTRN + "(" + ID_PTRN + ")\\s*(?:(?:={1,2})|(!=))\\s*(" + ID_PTRN + ")" + END_PTRN) {
-            @Override
-            public WhereFilter getExpression(String expression, Matcher matcher, Object... args) {
-                final String columnName = matcher.group(1);
-                final boolean inverted = matcher.group(2) != null;
-                final String paramName = matcher.group(3);
-
-                final FormulaParserConfiguration parserConfiguration = (FormulaParserConfiguration) args[0];
+                final boolean mirrored = !leftIsId;
```

Comment:
We're allowing literal op columName, but we're not allowing QueryScopeParam op columName. That feels incomplete, but the change might be pretty annoying. Might need more test coverage, as well.